arcgis-ios-sdk-samples/Utility network/Display content of utility network container/DisplayContentUtilityNetworkContainerViewController.swift

```diff
@@ -0,0 +1,360 @@
+            guard let self = self, let associations = containmentAssociations else { return }
+            let contentElements: [AGSUtilityElement] = associations.map { association in
+                if association.fromElement.objectID == containerElement.objectID {
+                    return association.toElement
+                } else {
+                    return association.fromElement
+                }
+            }
+            // If there are elements, find their corresponding features.
+            if !contentElements.isEmpty {
+                self.addGraphicsToFeatures(within: containerElement, for: contentElements)
+            }
+        }
+    }
+    
+    /// Get the features that correspond to the content's elements and add graphics.
+    ///
+    /// - Parameters:
+    ///     - containerElement: The container element of the selected feature
+    ///     - contentElements: An array of the content elements.
+    func addGraphicsToFeatures(within containerElement: AGSUtilityElement, for contentElements: [AGSUtilityElement]) {
+        // Save the previous viewpoint.
+        previousViewpoint = mapView.currentViewpoint(with: .boundingGeometry)
+        // Hide the layers to display the container view.
+        (mapView.map?.operationalLayers as? [AGSLayer])?.forEach { layer in
+            layer.isVisible = false
+        }
+        // Get the corresponding features for the array of content elements.
+        utilityNetwork?.features(for: contentElements) { [weak self] (contentFeatures, error) in
+            guard let self = self else { return }
+            if let contentFeatures = contentFeatures {
+                // Create and add a symbol for each of the features.
+                contentFeatures.forEach { content in
+                    let symbol = (content.featureTable as? AGSArcGISFeatureTable)?.layerInfo?.drawingInfo?.renderer?.symbol(for: content)
+                    self.graphicsOverlay.graphics.add(AGSGraphic(geometry: content.geometry, symbol: symbol))
+                }
+                // Set the bounding box which defines the container view may be computed using the extent of the features it contains or centered around its geometry at the container's view scale.
+                if contentFeatures.count == 1,
+                   let point = contentFeatures.first?.geometry as? AGSPoint {
+                    self.mapView.setViewpointCenter(point, scale: containerElement.assetType.containerViewScale) { _ in
+                        self.presentAlert(title: nil, message: "This feature contains no associations.")
+                        guard let boundingBox = self.mapView.currentViewpoint(with: .boundingGeometry)?.targetGeometry else { return }
+                        self.identifyAssociationsWithExtent(boundingBox: boundingBox)
+                    }
+                } else {
+                    let boundingBox = AGSGeometryEngine.bufferGeometry(self.graphicsOverlay.extent, byDistance: 0.05)
```

Comment:
This value is buffered again in `identifyAssociationsWithExtent(boundingBox:)`. Does it need to be buffered twice? It seems to me that the single point extent (in the other branch of the `if` statement) doesn't need to be buffered, so the extent could just be buffered here.

Reply:
The first buffers within the bounding box (yellow dashed line) and the second buffers outside of it. Without one or the other, symbols might be on the bounding box line or the bounding box line will be on the edge of the screen. IMO, both are necessary but there may be a more efficient solution. Do you have any further suggestions?

Reply:
I think I understand. Thanks!